### PR TITLE
Update 176 and ATB archivers to not hardcode year

### PR DESCRIPTION
**Pull request by e-belfer**
# Overview

Addresses part of #372.

What problem does this address?
With hardcoded years, we currently have to remember to check whether a new year of data is out. The monthly archiver runs won't capture new years of data released.

What did you change in this PR?
Removes hardcoded year from NREL ATB and EIA 176 archivers. 

# Testing

How did you make sure this worked? How can a reviewer verify this?
Run both archivers. Verify all years expected to download are successfully downloaded.

# To-do list

```[tasklist]
- [x] Update relevant documentation - like comments, docstrings, README, release notes, etc.
- [x] Review the PR yourself and call out any questions or issues you have
```

### CODE REVIEW

File: src/pudl_archiver/archivers/eia/eia176.py [hunk 1]
```diff
@@ -41,7 +42,11 @@ async def get_items(self, url: str = items_url) -> list[str]:
     async def get_resources(self) -> ArchiveAwaitable:
         """Download EIA 176 resources."""
         items_list = await self.get_items(url=self.items_url)
-        for year in range(1997, 2023):
+        datasets_list = await self.get_datasets(url=self.base_url, form=self.form)
+        dataset_years = set()
+        for dataset in datasets_list:
+            dataset_years.update([year.ayear for year in dataset.available_years])
```
Comment (zaneselvans) [inline src/pudl_archiver/archivers/eia/eia176.py:48]: So this is getting a list of available years for the dataset from the API?  At first glance I thought maybe it was looking at the years specified in the data source metadata "available years".  A comment or note in the docstring would help avoid that confusion.

File: src/pudl_archiver/archivers/eia/eia176.py [hunk 2]
```diff
@@ -41,7 +42,11 @@ async def get_items(self, url: str = items_url) -> list[str]:
     async def get_resources(self) -> ArchiveAwaitable:
         """Download EIA 176 resources."""
         items_list = await self.get_items(url=self.items_url)
-        for year in range(1997, 2023):
+        datasets_list = await self.get_datasets(url=self.base_url, form=self.form)
```
Comment (zaneselvans) [inline src/pudl_archiver/archivers/eia/eia176.py:45]: Should this line still be referring to `self.base_url` which has been removed above?

**Review by zaneselvans — CHANGES_REQUESTED** [review]
* Request for a bit more info in docstring.
* Question about maybe a variable name that needs to be updated?

File: src/pudl_archiver/archivers/eia/eia176.py (hunk 2 above)
Comment (e-belfer) [reply src/pudl_archiver/archivers/eia/eia176.py:45]: `base_url` is defined in the `EiaNGQVArchiver` class, it is the correct link here. I've updated the docstring to clarify.

File: src/pudl_archiver/archivers/eia/eia176.py (hunk 1 above)
Comment (e-belfer) [reply src/pudl_archiver/archivers/eia/eia176.py:48]: Noted, I've added a clarifying string.

**Review by zaneselvans — APPROVED** [review]

**Pull request merged**